lib/models/obms.js

```diff
@@ -208,6 +208,22 @@ function ObmsModelFactory (
             });
         },
 
+        findAllByNode: function(nodeId, reveal, query) {
+            var self = this;
+
+            query = query || {};
+            return Promise.try(function() {
+                    _.merge(query, {node: nodeId} );
+                    return self.find(query);
+                })
+                .then(function(obms) {
+                    if (obms && reveal) {
+                        return obms.map(_revealSecrests);
```

Comment:
The function name is not correct, it is `_revealSecrets` rather than your `_revealSecrests`